subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/RelativePathTracker.java

```diff
@@ -60,13 +62,31 @@ public boolean isRoot() {
         return segments;
     }
 
+    public String toAbsolutePath() {
+        if (segments.isEmpty()) {
+            return rootName;
+        } else {
+            int length = rootName.length() + segments.size();
+            for (String segment : segments) {
+                length += segment.length();
+            }
+            StringBuilder buffer = new StringBuilder(length);
+            buffer.append(rootName);
+            for (String segment : segments) {
+                buffer.append(File.separatorChar);
```

Comment:
We must use `/` as separator for relative paths, or the build cache key for relative path snapshots is not the same between Windows and Linux.